cni/pkg/nodeagent/informers.go

```diff
@@ -207,10 +207,16 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 		}
 		wasAnnotated := oldPod.Annotations != nil && oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
+		isOpOut := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionDisabled
+
```

Comment:
Why add this, when

`shouldBeEnabled := util.PodRedirectionEnabled(ns, newPod)`

already checks for `constants.AmbientRedirectionDisabled` in the line right after?

Reply:
Oh I see.

TY for finding and adding tests! We can actually simplify this a bit by weakening the `isAnnotated` check to just check for either label: https://github.com/istio/istio/pull/49209